newsroom/agenda/agenda.py

```diff
@@ -315,13 +310,40 @@ def _agenda_query():
     }
 
 
+def get_date_filter_query(filter_name: str):
+    """Get the query for the given filter name."""
+    time_filters = app.config.get("AGENDA_TIME_FILTERS", [])
+    if filter_name == "custom_date":
+        return None
+    for time_filter in time_filters:
+        if time_filter["filter"] == filter_name:
+            query = time_filter["query"]
+            return query
+    return None
+
+
 def get_date_filters(args):
     date_range = {}
     offset = int(args.get("timezone_offset", "0"))
-    if args.get("date_from"):
-        date_range["gt"] = get_local_date(args["date_from"], "00:00:00", offset)
-    if args.get("date_to"):
-        date_range["lt"] = get_end_date(args["date_to"], get_local_date(args["date_to"], "23:59:59", offset))
+
+    agenda_filter = get_date_filter_query(args.get("date_filter"))
+    date_from = args.get("date_from")
+
+    if agenda_filter:
+        if date_from:
+            start_date = get_local_date(date_from, "00:00:00", offset)
+            if agenda_filter.startswith("now-"):  # Handle past filters (e.g., last_7_days)
+                date_range["gt"] = get_local_date(agenda_filter, "00:00:00", offset)
+                date_range["lt"] = start_date
+            elif agenda_filter.startswith("now+"):  # Handle future filters (e.g., next_7_days)
```

Comment:
default filter we send from UI

Reply:
we don't send it anymore, that was causing issues in the UI

Reply:
but what I mean is rather here:
```py
    {"name": lazy_gettext("Today"), "query": "now/d", "filter": "today"},
    {"name": lazy_gettext("This Week"), "query": "now/w", "filter": "this_week"},
    {"name": lazy_gettext("This Month"), "query": "now/M", "filter": "this_month"},
```
because it seems like those queries are not covered, but we use them as defaults

Reply:
>we don't send it anymore; that was causing issues in the UI
I have made the changes in this PR to send default `date_filter`

Reply:
I see, but why? the server knows which one is the default

Reply:
added the case which handles these default filters, also removed functionality to send default `date_filter`